frontend-api/src/test/java/uk/gov/di/authentication/frontendapi/lambda/SendNotificationHandlerTest.java

```diff
@@ -168,6 +168,32 @@ void shouldReturn204AndPutMessageOnQueueForAValidVerifyEmailRequest()
         verify(sessionService).save(argThat(this::isSessionWithEmailSent));
     }
 
+    @Test
+    void shouldUseExistingOtpCodeIfOneExists() {
+        usingValidSession();
+        usingValidClientSession(CLIENT_ID);
+
+        when(codeStorageService.getOtpCode(any(String.class), any(NotificationType.class)))
+                .thenReturn(Optional.of(TEST_SIX_DIGIT_CODE));
+
+        APIGatewayProxyRequestEvent event = new APIGatewayProxyRequestEvent();
+        event.setHeaders(Map.of("Session-Id", session.getSessionId()));
+        event.setBody(
+                format(
+                        "{ \"email\": \"%s\", \"notificationType\": \"%s\", \"phoneNumber\": \"%s\" }",
+                        TEST_EMAIL_ADDRESS, VERIFY_PHONE_NUMBER, TEST_PHONE_NUMBER));
+        APIGatewayProxyResponseEvent result = handler.handleRequest(event, context);
+
+        verify(codeGeneratorService, never()).sixDigitCode();
+        verify(codeStorageService, never())
+                .saveOtpCode(
+                        any(String.class),
+                        any(String.class),
+                        any(long.class),
+                        any(NotificationType.class));
+        assertEquals(204, result.getStatusCode());
```

Comment:
This might be better for checkign the status code `assertThat(result, hasStatus(204))`